Sources/CheckoutNetwork/CheckoutNetworkClient+AsyncWrappers.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  CheckoutNetworkClient+AsyncWrappers.swift
+//
+//
+//  Created by Okhan Okbay on 20/02/2024.
+//
+
+import Foundation
+
+public extension CheckoutNetworkClient {
+
+  func runRequest<T: Decodable>(with configuration: RequestConfiguration) async throws -> T {
+    return try await withCheckedThrowingContinuation { continuation in
+      runRequest(with: configuration) { (result: Result<T, Error>) in
+        switch result {
+        case .success(let response):
+          continuation.resume(returning: response)
+        case .failure(let error):
+          continuation.resume(throwing: error)
+        }
+      }
+    }
+  }
+
+  func runRequest(with configuration: RequestConfiguration) async throws {
+    return try await withCheckedThrowingContinuation { continuation in
+      runRequest(with: configuration) { (error: Error?) in
```

Comment:
same. it can't infer the type when it's removed so that it's necessary. Even when not, it's faster for compiler to have a defined type rather than let it infer thereof.